endpoints/openrtb2/auction.go

```diff
@@ -2352,6 +2352,33 @@ func setDoNotTrackImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrappe
 	}
 }
 
+func secCookieDeprecation(httpReq *http.Request, r *openrtb_ext.RequestWrapper) {
+	secCookieDeprecation := httpReq.Header.Get("Sec-Cookie-Deprecation")
+	if secCookieDeprecation == "" {
+		return
+	}
+
+	if r.Device == nil {
+		r.Device = &openrtb2.Device{}
+	}
+
+	updated := false
+	deviceExt := map[string]interface{}{}
+	if r.Device.Ext == nil {
+		updated = true
+		deviceExt["cdep"] = secCookieDeprecation
+	} else {
+		if err := json.Unmarshal(r.Device.Ext, &deviceExt); err == nil && deviceExt["cdep"] == nil {
+			updated = true
+			deviceExt["cdep"] = secCookieDeprecation
+		}
+	}
+
+	if updated {
+		r.Device.Ext, _ = json.Marshal(deviceExt)
+	}
+}
```

Comment:
This should use the RequestWrapper instead of direct json manipulation. There should be no Unmarshal or Marshal calls in this function. You will need to add `cdep` to `DeviceExt`.

Reply:
done